extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonResource.java

```diff
@@ -112,7 +112,7 @@ public Dog echoDog(Dog dog) {
     @POST
     @Path("/record-echo")
     @Consumes(MediaType.APPLICATION_JSON)
-    public StateRecord echoDog(StateRecord stateRecord) {
+    public StateRecord echoRecord(StateRecord stateRecord) {
```

Comment:
Are you perhaps missing something from the commit?
Asking because I don't see how this exercises the scenario this change is supposed to address

Reply:
This method renaming is unrelated, it is just to fix the formerly wrong name that came from an excess of copy/paste :)

This endpoint and the test invoking it are actually exercising this improvement in the sense that returned `StateRecord` wasn't deserialized through the generated reflection-free deserializer before, since the record structurally doesn't have an no-args constructor. Now it generates the deserializers also for records. The final effect is of course the same (of course in a reflection-free way this time), the only issue is that I don't know how to check that the deserializers is actually generated and it is using it, since the net behaviour is the same.

Reply:
Ok, thanks.

I don't have any suggestions for better testing at the moment